File: surf-cloud-standalone/src/main/java/dev/slne/surf/cloud/standalone/__AdditionalSpringStandaloneConfiguration.java
package dev.slne.surf.cloud.standalone;

import dev.slne.surf.cloud.api.meta.SurfJpaRepository;
import dev.slne.surf.cloud.api.meta.SurfRedisRepository;
import org.springframework.context.annotation.ComponentScan.Filter;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.FilterType;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.data.redis.repository.configuration.EnableRedisRepositories;

@EnableJpaRepositories(
    includeFilters = @Filter(type = FilterType.ANNOTATION, classes = SurfJpaRepository.class),
    excludeFilters = @Filter(type = FilterType.ANNOTATION, classes = SurfRedisRepository.class),
    repositoryImplementationPostfix = "JpaImpl"
)
@EnableRedisRepositories(
    includeFilters = @Filter(type = FilterType.ANNOTATION, classes = SurfRedisRepository.class),
    excludeFilters = @Filter(type = FilterType.ANNOTATION, classes = SurfJpaRepository.class),
    repositoryImplementationPostfix = "RedisImpl"
)
@Configuration
class __AdditionalSpringStandaloneConfiguration {

}

File: surf-cloud-standalone/src/main/kotlin/dev/slne/surf/cloud/standalone/__AdditionalSpringStandaloneConfiguration.kt
package dev.slne.surf.cloud.standalone

import dev.slne.surf.cloud.api.server.plugin.AdditionalStandaloneConfiguration
import dev.slne.surf.cloud.api.server.plugin.TransactionConfiguration
import org.jetbrains.exposed.spring.autoconfigure.ExposedAutoConfiguration
import org.springframework.boot.autoconfigure.ImportAutoConfiguration
import org.springframework.boot.autoconfigure.data.redis.RedisAutoConfiguration
import org.springframework.context.annotation.Import

/**
 * Additional Spring configuration for standalone applications,
 * enabling selective JPA and Redis repository scanning.
 */
@Suppress("ClassName")
@AdditionalStandaloneConfiguration
@ImportAutoConfiguration(ExposedAutoConfiguration::class, RedisAutoConfiguration::class)
@Import(TransactionConfiguration::class)
//@EnableAspectJAutoProxy(proxyTargetClass = true)
internal class __AdditionalSpringStandaloneConfiguration
